analyzer/tests/unit/test_checker_handling.py

```diff
@@ -272,14 +338,18 @@ def test_disable_clangsa_checkers(self):
 
         for arg in analyzer.construct_analyzer_cmd(result_handler):
             if arg.startswith('-checks='):
-                self.assertIn('-clang-analyzer-*', arg)
+                self.assertTrue(self._is_disabled(
+                    'clang-analyzer',
+                    arg[len('-checks='):].split(',')))
```

Comment:
Yeah, this isn't any better. It should look like this:
```python
self.assertTrue(self._is_disabled('clang-analyzer', arg))
```
and other nasties should be inside `_is_disabled`. Do you see my point?

Reply:
If I do it this way then `test_default_checkers_are_not_disabled()` and `test_clangsa_analyzer_checks_are_disabled()` will be ugly. Which one should we save? :)

Reply:
Done, restructured.